compiler/qsc/src/interpret.rs

```diff
@@ -639,6 +672,45 @@ impl Interpreter {
         })
     }
 
+    /// Performs QIR codegen using the given callable with the given arguments on a new instance of the environment
+    /// and simulator but using the current compilation.
+    pub fn qirgen_from_callable(
+        &mut self,
+        callable: &Value,
+        args: Value,
+    ) -> std::result::Result<String, Vec<Error>> {
+        if self.capabilities == TargetCapabilityFlags::all() {
+            return Err(vec![Error::UnsupportedRuntimeCapabilities]);
```

Comment:
Does QIR gen only work for Base profile still? If so, is checking for all target capabilities correct, or should we check for `!= Base`?

Reply:
No, this case is correct because we only reject if all capabilities are asked for. We can support Base and two flavors of Adaptive now.